src/Native/Unix/System.Native/pal_time.c

```diff
@@ -48,75 +48,43 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
     return result;
 }
 
-int32_t SystemNative_GetTimestampResolution(uint64_t* resolution)
+uint64_t SystemNative_GetTimestampResolution()
 {
-    assert(resolution);
-
 #if HAVE_MACH_ABSOLUTE_TIME
     mach_timebase_info_data_t mtid;
-    if (mach_timebase_info(&mtid) == KERN_SUCCESS)
-    {
-        *resolution = SecondsToNanoSeconds * ((uint64_t)(mtid.denom) / (uint64_t)(mtid.numer));
-        return 1;
-    }
-    else
-    {
-        *resolution = 0;
-        return 0;
-    }
 
-#elif HAVE_CLOCK_MONOTONIC
+    kern_return_t result = mach_timebase_info(&mtid);
+    assert(result == KERN_SUCCESS);
+    (void)result; // suppress unused parameter warning in release builds
+
+    return SecondsToNanoSeconds * ((uint64_t)(mtid.denom) / (uint64_t)(mtid.numer));
+#else
     // Make sure we can call clock_gettime with MONOTONIC.  Stopwatch invokes
     // GetTimestampResolution as the very first thing, and by calling this here
     // to verify we can successfully, we don't have to branch in GetTimestamp.
-    struct timespec ts;
-    if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) 
-    {
-        *resolution = SecondsToNanoSeconds;
-        return 1;
-    }
-    else
-    {
-        *resolution = 0;
-        return 0;
-    }
 
-#else /* gettimeofday */
-    *resolution = SecondsToMicroSeconds;
-    return 1;
+    struct timespec ts;
 
+    int result = clock_gettime(CLOCK_MONOTONIC, &ts);
+    assert(result == 0);
+    (void)result; // suppress unused parameter warning in release builds
+    
+    return SecondsToNanoSeconds;
 #endif
 }
 
-int32_t SystemNative_GetTimestamp(uint64_t* timestamp)
+uint64_t SystemNative_GetTimestamp()
 {
-    assert(timestamp);
-
 #if HAVE_MACH_ABSOLUTE_TIME
-    *timestamp = mach_absolute_time();
-    return 1;
-
-#elif HAVE_CLOCK_MONOTONIC
```

Comment:
Every Unix platform we care to target has either mach_absolute_time or clock_gettime(CLOCK_MONOTONIC, …) ?

Reply:
We assume `CLOCK_MONOTONIC` for ARM/ARM64 (https://github.com/dotnet/corefx/blob/b3b1777642c80f7a0ba57b32581c33442ceac1d7/cross/android/arm64/tryrun.cmake#L4) and I think it is reasonable to say that any system that doesn't have it is unsupported.

Reply:
It is technically optional according to the POSIX specification, however.